xtask/src/main.rs

```diff
@@ -334,6 +337,47 @@ fn run_example(workspace: &Path, mut args: RunExampleArgs) -> Result<()> {
     Ok(())
 }
 
+fn build_tests(workspace: &Path, args: RunTestsArgs) -> Result<(), anyhow::Error> {
+    // Absolute path of the package's root:
+    let package_path = xtask::windows_safe_path(&workspace.join("hil-test"));
+
+    // Determine the appropriate build target for the given package and chip:
+    let target = target_triple(&Package::HilTest, &args.chip)?;
+
+    // Load all examples and parse their metadata:
+    let tests = xtask::load_examples(&package_path.join("tests"))?;
+    let mut supported_tests = tests
+        .iter()
+        // Filter down the examples to only those for which the specified chip is supported:
+        .filter(|example| example.supports_chip(args.chip));
+    if let Some(test_name) = &args.test {
+        let test = supported_tests.find_map(|example| {
+            if &example.name() == test_name {
+                Some(example.clone())
+            } else {
+                None
+            }
+        });
+        if let Some(test) = test {
+            xtask::build_example(&package_path, args.chip, target, &test)?;
+        } else {
+            log::error!("Test not found or unsupported for the given chip");
+        }
+    } else {
+        let mut failed_tests: Vec<String> = Vec::new();
+        for test in supported_tests {
+            if xtask::build_example(&package_path, args.chip, target, test).is_err() {
+                failed_tests.push(test.name());
+            }
+        }
+        if !failed_tests.is_empty() {
+            bail!("Failed tests: {:?}", failed_tests);
+        }
+    }
+
+    Ok(())
+}
```

Comment:
This code could be deduplicated as currently we have `build_examples`, `run_example`, `build_tests`, `run_tests`

Reply:
yeah the xtask could do a with a bit a refactoring, but I'm not going to lose sleep over this right now :D.

Reply:
> yeah the xtask could do a with a bit a refactoring, but I'm not going to lose sleep over this right now :D.

If we keep this attitude it's never going to get done, IMO.